web/source/keymanweb.js

```diff
@@ -3681,21 +3722,30 @@
     } else {
       if(Pelem.tagName.toLowerCase() != 'iframe') {
         if(keymanweb.isKMWInput(Pelem)) {
-          var x = keymanweb.setupTouchElement(Pelem);
-
-          if(x) {
-            // if needed, let the timeout variable be: keymanweb.touchInputAddedTimer
-            window.setTimeout(function() {
-              keymanweb.updateInput(x);
-            }, 1);
-          }
+          keymanweb.setupTouchElement(Pelem);
         } else {
           keymanweb.setupNontouchElement(Pelem);
         }
       } // no 'else' - the touch-device implementation never handled iframes.
     }
   }
 
+  // Used by the mutation event handler to properly decouple any elements dynamically removed from the document.
+  keymanweb._MutationRemovalObserved = function(Pelem) {
+    var element = Pelem;
+    if(device.touchable) {
+      element = Pelem['kmw_ip'];
+      
+      // We get weird repositioning errors if we don't remove our simulated input element - and permanently.
+      Pelem.parentNode.removeChild(element);
+      Pelem['kmw_ip'] = null;  // Erase the evidence that this was ever here.
```

Comment:
Yes. You don't need to set to `null`; you do need to `removeChild` and `delete`.